chain-addr/src/lib.rs

```diff
@@ -74,6 +79,7 @@ impl KindType {
 
 /// An unstructured address including the
 /// discrimination and the kind of address
+#[cfg_attr(feature = "generic-serialization", derive(Serialize))]
```

Comment:
Nop, this is where it gets wrong. The address is not a generic serialisable. The serde implementation is only for the `AddressReadable` type. Not for the other types.

Reply:
I've changed `Serde` serialization of `Address` to reuse its `property::Serialize` implementation. Is it ok?